### ISSUE TITLE
Sending test emails in administration doesn't use template files

### ISSUE THREAD

**Issue by stephan-schuster** [FriendsOfShopware/FroshPlatformTemplateMail#34]
### PHP Version

7.4

### Shopware Version

6.4.10

### Actual behaviour

When I send a test email from Admin -> Settings-> Email templates it does not use the template from file system, but the ones from database. Same happens when clicking the preview icon.

### Expected behaviour

When the plugin is activated, preview and send test email functions should also use the templates from theme file system

### Steps to Reproduce?

1. Log in into shopwares admin area
2. go to Settings -> Email templates
3. choose Order Confirmation 
4. click an "send test email" symbol on the right
5. enter mail recipient, choose sales channel 
6. click send button

**Comment by shyim** [comment]
I guess I would rather remove the entire module using an option as I assume that people don't want that module anymore as templates defines it 🤔

**Comment by stephan-schuster** [comment]
> I guess I would rather remove the entire module using an option as I assume that people don't want that module anymore as templates defines it 🤔

I would like to continue using this feature as it is much more convenient than placing a new order, registering an account, etc. every time I make a change to the template.

**Comment by benjamin-kentsch** [comment]
People still want that test feature. Showed the plugin to my project manager today and the feature was missed right away. It is much more easier to use the test email than to generate the neccessary mail by registration or whatever.

**Comment by danielDigitalArt** [comment]
Hi,

i got it working for me, when i click on the test-mail send button i get the renderet Theme file instead of the database content. I replaced the contents in the Database with simple Theme Extend Snippets like so:

{% sw_extends "@myTheme/email/en-GB/order_confirmation_mail/html.twig" %}

![image](https://user-images.githubusercontent.com/25100967/203966158-7bf30028-0175-4498-8d82-ae3529cb53b2.png)

Don yet know if there will be a problem somewhere else, but maybe you want to give it a shot.

**Comment by mbuchwaldt** [comment]
Hi, 

Is there any news on this topic? It would be nice to have a preview option to see what the current template looks like. Everything else is a bit difficult.

Best regards

**Comment by niklaswolf** [comment]
I think we could get the preview-functionality working with the template-files, but we'd need a slight non-breaking change in the administration:
In the component that triggers the API-Call to `\Shopware\Core\Content\MailTemplate\Api\MailActionController::send` we'd need to provide the `mailTemplateId` and `mailTemplateTypeId` to the API. Thats [here](https://github.com/shopware/shopware/blob/9f9b3ec9a198a7c19e893f97fd0a53788e188ee9/src/Administration/Resources/app/administration/src/module/sw-mail-template/page/sw-mail-template-detail/index.js#L331)

We could then use some of the exisiting events, for example the `MailBeforeValidateEvent` to replace the templates with the content from the filesystem. This would more or less be the same logic as currently implemented in the `FlowSubscriber`, so I think we could just move the logic to a subscriber for the `MailBeforeValidateEvent`.

What do you guys think?
I'd be willing to tackle this further, of course the precondition is that we can get the change merged in the administration :)

**Comment by danielDigitalArt** [comment]
> 
> @danielDigitalArt Unfortunately, this does not work for me, as SW 6 interprets this as a syntax error.
> 
> Best regards

@mbuchwaldt is just retested this in SW 6.6.6.1 and it still worked. Only missing translated Values and Address-Data, but the template itself is loaded.

**Comment by BenRoe** [comment]
> Hi,
> 
> i got it working for me, when i click on the test-mail send button i get the renderet Theme file instead of the database content. I replaced the contents in the Database with simple Theme Extend Snippets like so:
> 
> {% sw_extends "@myTheme/email/en-GB/order_confirmation_mail/html.twig" %}
> 
@danielDigitalArt can you share the table name and entry i need to change?

**Comment by Schrank** [comment]
@niklaswolf I think on the call is everything we need (looking on Version 6.6.9.0 here):

```
{
    "contentHtml": "<div style=\"font-family:arial; font-size:12px;\">\n\n    {% if order.orderCustomer.salutation %}{{ order.orderCustomer.salutation.translated.letterName ~ ' ' }}{% endif %}{{ order.orderCustomer.firstName }} {{ order.orderCustomer.lastName }},<br>\n    <br>\n    Im Anhang dieser E-Mail finden Sie die Stornorechnung für Ihre Bestellung mit der Nummer: {{ order.orderNumber }}<br>\n    <br>\n    Sollten Sie Fragen haben, zögern Sie nicht, uns zu kontaktieren.\n    <br>\n</div>\n",
    "contentPlain": "{% if order.orderCustomer.salutation %}{{ order.orderCustomer.salutation.translated.letterName ~ ' ' }}{% endif %}{{ order.orderCustomer.firstName }} {{ order.orderCustomer.lastName }},\n\nIm Anhang dieser E-Mail finden Sie die Stornorechnung für Ihre Bestellung mit der Nummer: {{ order.orderNumber }}\n\nSollten Sie Fragen haben, zögern Sie nicht, uns zu kontaktieren.\n",
    "mailTemplateData": {
        "order": {
            "orderNumber": "10060",
            "orderCustomer": {
                "firstName": "Max",
                "lastName": "Mustermann"
            }
        },
        "salesChannel": {
            "name": "Storefront"
        }
    },
    "recipients": {
        "sad@example.com": "sad@example.com"
    },
    "salesChannelId": "fe6c20cd96a14619ad7cdd6110675fe6",
    "mediaIds": [],
    "subject": "New document for your order",
    "senderName": "{{ salesChannel.name }}",
    "documentIds": [],
    "testMode": true,
    "mailTemplateTypeId": "c7deebf5eb4b457baac3e7f917c7cf04",
    "mailTemplateId": "a08663ff9e0a466b892f86177e12bfd1"
}
```

**Comment by niklaswolf** [comment]
@Schrank you are right, because[ my PR](https://github.com/shopware/shopware/pull/4850) was merged :) 
Should be there since 6.6.7.0

So now we could progress with https://github.com/FriendsOfShopware/FroshPlatformTemplateMail/issues/34#issuecomment-2363260413
